### PR TITLE
Fix a typo extendable enumeration

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]

**Review by LarryOsterman — APPROVED** [review]

File: sdk/core/azure-core/inc/azure/core/internal/extendable_enumeration.hpp [hunk 1]
```diff
@@ -20,7 +20,7 @@ namespace Azure { namespace Core { namespace _internal {
    * Example:
    *
    * \code{.cpp}
-   * class MyEnumeration final : public ExtendableEnumeration<AttestationType> {
+   * class MyEnumeration final : public ExtendableEnumeration<MyEnumeration> {
    * public:
    *   explicit MyEnumeration(std::string attestationType) :
```
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/internal/extendable_enumeration.hpp:25]: Should we generalize other instances in the doc comments that are specific to just attestation, like this (and below), @LarryOsterman?

Maybe like `serviceEnumType`

**Review by ahsonkhan — APPROVED** [review]

File: sdk/core/azure-core/inc/azure/core/internal/extendable_enumeration.hpp (hunk 1 above)
Comment (LarryOsterman) [reply sdk/core/azure-core/inc/azure/core/internal/extendable_enumeration.hpp:25]: Agreed. THis was a sloppy C&P on my part. @Jinming-Hu would you mind doing that as well as a part of this pull request.

**Comment by LarryOsterman** [comment]
One other suggestion, while you're in the code.

This header doesn't have an include guard:

```c++
#if !defined(INCLUDE_AZURE_CORE_EXTENDABLE_ENUMERATION_HPP)
#pragma once

#define INCLUDE_AZURE_CORE_EXTENDABLE_ENUMERATION_HPP
<STUFF>
#endif // !defined(INCLUDE_AZURE_CORE_EXTENDABLE_ENUMERATION_HPP)


```

**Comment by Jinming-Hu** [comment]
@LarryOsterman I've updated the PR. Feel free to click the merge button.

As I see it, the constructors of base class should be protected because we never need to construct a base class directly. But doing that would be a breaking change, so I'll leave them as is.

**Pull request merged**